assets/javascripts/discourse/widgets/discourse-reactions-reaction-button.js.es6

```diff
@@ -60,17 +75,45 @@ export default createWidget("discourse-reactions-reaction-button", {
       title = "discourse_reactions.main_reaction.cant_remove";
     }
 
-    return { title: I18n.t(title) };
+    if (
+      currentUserReaction &&
+      currentUserReaction.can_undo &&
+      !likeAction.hasOwnProperty("can_undo")
```

Comment:
This is not something we do a lot in the codebase, it would be worth a comment to explain why you decided to use hasOwnProperty here